gcp/api/server.py

```diff
@@ -447,6 +454,10 @@ def do_query(query, context: QueryContext, include_details=True):
     ecosystem = ''
     purl_str = ''
 
+  if ecosystem and ecosystem not in valid_ecosystems():
```

Comment:
I think we could check `ecosystems.get()` for this instead of querying
https://github.com/google/osv.dev/blob/310c28d0a708f28fcabd96a0d335a278d3fde549/osv/ecosystems/_ecosystems.py#L66

I'm pretty sure we only add things to datastore if we have helpers for them

Reply:
Yeah, that works for me and is far easier to implement, thanks for the suggestion. Implemented in https://github.com/google/osv.dev/pull/1485/commits/5c4385bcb5599476c4fa0bab0a2254464e787942 with test coverage in https://github.com/google/osv.dev/pull/1485/commits/36ae72269fb7f22d992736f0f0ad7cd2757a662b and https://github.com/google/osv.dev/pull/1485/commits/4c790de0e8ad8fd627ca9645b7b790be2fc771b5